lib/matplotlib/style/__init__.py

```diff
@@ -245,8 +245,9 @@ def update_nested_dict(main_dict, new_dict):
 def reload_library():
     """Reload the style library."""
     library.clear()
-    library.update(_update_user_library(_base_library))
-    available[:] = sorted(library.keys())
+    library.update(_update_user_library(_base_library.copy()))
+    available[:] = sorted(name for name in library
+                          if not name.startswith('_') or name not in _base_library)
```

Comment:
Ok, so the idea is that we only exclude "_name" styles that are in the base library. Any other styles in the base library are shown, and *any* user styles are shown. I got that right?

Reply:
I would l go for: Exclude all styles with leading _. Not sure why users would register such styles, but let’s just document this way. It’s more in line with general underscore semantics , e.g. `dir()` and simpler to implement.

Reply:
My take: Private styles are those with a leading underscore. Everybody can register and get/use them. They are only not listed under the available styles. This is the same logic as with private attributes/methods.

Reply:
Yes, the only reason I wrote it this way is because the test adds a `_test_` style and looks for it in `available`. But as I've now fixed `reload_library` to correctly clean up user styles, I think we can use any name in the test.